agent/api/task/task.go

```diff
@@ -1760,23 +1760,47 @@ func (task *Task) PopulateSecretsAsEnv(container *apicontainer.Container) *apier
 
 	envVars := make(map[string]string)
 
+	logDriverTokenName := ""
+	logDriverTokenSecretValue := ""
+
 	for _, secret := range container.Secrets {
-		if secret.Provider == apicontainer.SecretProviderSSM && secret.Type == apicontainer.SecretTypeEnv {
-			k := secret.GetSecretResourceCacheKey()
-			if secretValue, ok := ssmRes.GetCachedSecretValue(k); ok {
-				envVars[secret.Name] = secretValue
-			}
+		k := secret.GetSecretResourceCacheKey()
+
+		secretVal := ""
+		validValue := false
+
+		if secretValue, ok := ssmRes.GetCachedSecretValue(k); ok {
+			validValue = ok
+			secretVal = secretValue
+		}
+		if secretValue, ok := asmRes.GetCachedSecretValue(k); ok {
+			validValue = ok
+			secretVal = secretValue
 		}
 
-		if secret.Provider == apicontainer.SecretProviderASM && secret.Type == apicontainer.SecretTypeEnv {
```

Comment:
I don't think you should the provider condition for env secret and always try to get secret value from both.. ssmRes and asmRes will not always be initialized